src/chapters/control-flow-and-collections/loops.rst

```diff
@@ -0,0 +1,323 @@
+statement within the innermost loop, then it will only terminate that
+loop and not the outer one. If a ``break`` is present in the outer loop,
+it --- and any other block nested within it --- is terminated when the
+``break`` runs.
+
+.. index:: ! continue
+
+Continue Statements in Loops
+----------------------------
+
+The **continue** statement is similar to, but importantly different
+from, the ``break`` statement. Like ``break``, it interrupts the normal
+flow of control of the loop. But unlike ``break``, the ``continue``
+statement only terminates the *current iteration* of the loop. So the
+loop will continue to run from the top after a ``continue``
+statement. Here is an example:
+
+.. sourcecode:: csharp
+   :linenos:
+
+   int[] someInts = {1, 10, 2, 3, 5, 8, 10};
+   int searchTerm = 10;
+   foreach (int oneInt in someInts) {
+      if (oneInt == searchTerm) {
+         Console.WriteLine("Found it!");
+         continue;
+      }
+      Console.WriteLine("Not here");
+   }
+
+The above program will print "Not here" on every iteration of the
+``for`` loop *except* where the number has been found. So the output
+looks like this:
+
+.. sourcecode:: bash
+
+   Not here
+   Found it!
+   Not here
+   Not here
+   Not here
+   Not here
+   Found it!
+
+Because of the ``continue`` statement, the final print statement in the
+for loop is skipped. If the ``continue`` statement weren’t there, the
+output would look like this instead (notice the extra "Not here"
+printouts):
+
+.. sourcecode:: bash
+
+   Not here
+   Found it!
+   Not here
+   Not here
+   Not here
+   Not here
+   Not here
+   Found it!
+   Not here
+
+Concatenating Strings in Loops
+------------------------------
+
+A use case of loops may be to concatenate strings.
+For example, if we have a list of strings containing each word in a sentence, we may want to concatenate each value in the list to reform our sentence.
+
+.. sourcecode:: csharp
+   :linenos:
+
+   string finalSentence = "";
+
+   foreach (string i in listOfWords) {
+      finalSentence += i;
+   }
+
+This code would work well for this situation.
+However, we can accomplish the same thing with the ``StringBuilder`` class.
+``StringBuilder`` objects are *mutable* strings of characters and the `documentation <https://docs.microsoft.com/en-us/dotnet/api/system.text.stringbuilder?view=netframework-4.8>`_ contains a full list of important properties and methods.
+
+If we wanted to use a ``StringBuilder`` object instead of a simple string in the above code, we would modify it like so:
+
+.. sourcecode:: csharp
+   :linenos:
+
+   StringBuilder finalSentence = new StringBuilder();
+
+   foreach (string i in listOfWords) {
+      finalSentence.Append(i);
+   }
+
+First, we need to initialize a new ``StringBuilder`` object, ``finalSentence``, with ``new StringBuilder()``.
+The ``Append()`` method in the ``StringBuilder`` class adds the value of ``i`` to the end of the ``finalSentence`` object.
```

Comment:
It is supposed to be the value of ``i`` because ``i`` is not a counter variable, but is going through the values in the array. I changed the variable name to ``word`` to eliminate confusion.